### ISSUE TITLE
FR: CPU Overview on all modules / CPU Auto-Suspend (Smart CPU Saver)

### ISSUE THREAD

**Issue by Lykdraft** [kmatheussen/radium#672]
Copied from another thread:

**First:** I wonder if it might be a good idea to have an option to show the **CPU numbers 
superimposed over every module in the mixer**, so that one doesn't have to go through 80+ or 100+ modules to take a look if some problems arise. (My average module count on the last 4-6 tracks is pretty consistently around 80-90 modules.) Also this might be a good way to quickly get an overview. (Or maybe at least the average CPU number)

I could imagine just a checkbox besides theese arrow-icons that rotate/flip the mixer.
Not sure how much trouble this would be to realize though.

**Second:** a CPU Management Feature. Would help a lot when hitting limits and without a Freeze-Function: Auto-Suspend (or Smart CPU saving)
Simply put: Auto suspend **is putting a third-party plugin off when its not in use.**

_Bitwig does this pretty well, MuLab has something like this also. They both rely on this as both of them also do not have a Freeze-Function like most other DAWs. As does the plugin host/chainer Nyrv Agentl. (Called smart-cpu-saver there ) (smart cpu saver in Nyrv Agent: https://www.youtube.com/watch?v=j12KdIiK8q8))_

**Comment by kmatheussen** [comment]
Implemented auto-suspend. I called it auto-bypass, but auto-suspend is much better. Will rename in the next release.

**Comment by Lykdraft** [comment]
Crazy... here we are already.. Now let's see how this works. I'll test this with the CPU-limit-hitting tracks I did recently later and report back.

**Comment by Lykdraft** [comment]
is it correct that auto-bypass right now only gets triggered with the sampler instrument?
It doesn't seem to bypass other VST instruments (tested NI Massive just now) and also no effects?

(Sampler Instrument seems to be the only one where auto-bypass kicks in.)

**Comment by Lykdraft** [comment]
(native lasdpa and pipe gets autobypassed as well. )

**Comment by kmatheussen** [comment]
Seems like auto-suspending doesn't work for VST plugins. I added extra time if the VST plugin asks for timing information (to avoid losing sync), but the JUCE library asks for timing information for every audio block whether the VST plugin needs it or not. Don't know how to solve that yet. Everything else should work, except for system out/system in/jack in/jack out which needs to process no matter what.

**Comment by Lykdraft** [comment]
Alright, that's a very good start though anyways... :)

**Comment by Lykdraft** [comment]
Great, it's all done here.
